include/LightGBM/utils/openmp_wrapper.h

```diff
@@ -5,6 +5,10 @@
 #ifndef LIGHTGBM_OPENMP_WRAPPER_H_
 #define LIGHTGBM_OPENMP_WRAPPER_H_
 
+/** In an external multi-threaded environment, the below methods can return different values on
```

Comment:
"below methods" mainly refers to the ones with "max" in the name.  I will edit that.

It appears that by default, OpenMP sets OMP_THREAD_LIMIT to 1024. 
https://docs.oracle.com/cd/E37069_01/html/E37081/aewbg.html#scrolltoc

I guess that's really what we want, the OpenMP thread limit. I was just trying to use something smaller since I doubt many scenarios would hit that, but you are correct, we never know.

Users can set OMP_THREAD_LIMIT to something else (although of course hardware limitations apply). Would you like me to try to get the env var and then use 1024 if not set? Or is it ok to use 1024? (we'll just create a lot of empty buffers for 99.999% of cases likely). We could also try to use the hardware limit (likely to be lower), but most web searches show that's a hard thing to get as a constant. And I'm not sure how to handle the GPU case.

I'd suggest going with the env var or 1024 if not set, if that's ok.

Reply:
> I'd suggest going with the env var or 1024 if not set, if that's ok.

Yes, I think we should do that.

Reply:
see discussion below with @guolinke.  Went with a separate API param rather than a constant.